addons/website/migrations/12.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_field_renames = [
+    ('website.page', 'website_page', 'website_published', 'is_published'),
+]
+
+
+def fill_website_company_id(cr):
+    # done in pre-migration to avoid warning messages
+    openupgrade.logged_query(
+        cr, """
+        UPDATE website w
+        SET company_id = c.id
+        FROM (SELECT id FROM res_company ORDER BY id ASC LIMIT 1) AS c
+        WHERE w.company_id IS NULL
+        """
+    )
+
+
+def fill_website_name(cr):
+    # done in pre-migration to avoid warning messages
+    openupgrade.logged_query(
+        cr, """
+        UPDATE website w
+        SET name = 'Home #' || w.id
+        WHERE w.name IS NULL
+        """
+    )
+
+
+def fill_website_redirect_type(cr):
+    # done in pre-migration to avoid warning messages
+    openupgrade.logged_query(
+        cr, """
+        UPDATE website_redirect wr
+        SET type = '301'
+        WHERE wr.type IS NULL
+        """
+    )
+
+
+def fill_website_redirect_urls(cr):
+    # done in pre-migration to avoid warning messages
+    openupgrade.logged_query(
+        cr, """
+        UPDATE website_redirect wr
+        SET active = FALSE
+        WHERE wr.url_from IS NULL OR wr.url_to IS NULL
+        """
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    cr = env.cr
+    openupgrade.rename_fields(env, _field_renames)
+    fill_website_company_id(cr)
+    fill_website_name(cr)
+    fill_website_redirect_type(cr)
+    fill_website_redirect_urls(cr)
+    openupgrade.delete_records_safely_by_xml_id(
+        env, [
+            'website.aboutus',
+            'website.s_image_text_default_image',
+            'website.main_menu',
+            'website.s_cover_default_image',
+            'website.default_website',
+            'website.menu_contactus',
```

Comment:
Why removing this? Are you using this technique instead of loading the noupdate records for website.menu? It's better to update the records due to possible references to this record.

Reply:
But you said me before that you wanted to delete the records instead to update them in `noupdate_changes.xml` 😩

Reply:
No, I said to remove that records on the noupdate_records.xml file, not to remove them on Odoo!

Reply:
Any way, due to the investigation of website views duplication, if you duplicate views before loading noupdate records, we can keep the file as is.

Reply:
I am so confused now. Tell what should I do clearly.

Reply:
1. Duplicate QWeb views for each website, being one or several websites.
2. Load noupdate_records.xml as is